### ISSUE TITLE
"sequence not ordered" after building transmission on Windows

### ISSUE THREAD

**Issue by muyunmiao** [transmission/transmission#7182]
### What is the issue?

I what to compile the source code of version 4.0.4 on Windows 11. However, there's some errors occured when I finish the steps on "Building-Transmission.md" for Windows. It says that "sequence not ordered" in some file of  Visual Studio. I tried to use different versions of Visual Studio, which is no use. I also tried transmission-daemon in command line, which can not use, neither. How can I handle this problem? Or is there any other problems in these part?
![d66275e44f2d062568b888b98566efe](https://github.com/user-attachments/assets/60de790b-ce86-4343-ba4c-d249ee30b1db)

### Which application of Transmission?

Qt app on Windows

### Which version of Transmission?

4.0.4

**Comment by killemov** [comment]
You are running a **debug** build that fails on assertions. An assertion is a statement that makes sure something is true and breaks the program if it s not. You could run it from within Visual Studio or build a **release** version.

**Comment by ckerr** [comment]
I agree with @killemov, I don't think there's enough information here to debug the issue.

If you just want to use Transmission, download an official build or build it yourself in release mode.

If you want to help track down the assertion failure, run it in Visual Studio and get a backtrace of the code that led up to the assertion failure. According to https://stackoverflow.com/questions/23318981/expression-sequence-not-ordered this can happen when the ranges passed into `std::set_intersection()` are not sorted; but the only place I see that being called directly in Transmission is in the details dialog, and (a) we explicitly sort before making the call and (b) I don't see the details dialog in that screrenshot, so that doesn't seem to be the call that's causing the assertion failure.

**Comment by killemov** [comment]
@Coeur, @ckerr Is there any pr or commit associated with this or is this just silently closed because of reasons? (And THEN assign @muyunmiao to this.) :confused:

**Comment by Coeur** [comment]
Closing because no actions can be taken.
If you reproduce, then please provide reproduction steps or a debug backtrace.

**Comment by mikedld** [comment]
I'd say this could also be issued in other places which expect a sorted sequence, like e.g. `std::binary_search`, `std::lower_bound`, `std::upper_bound`, and there's more than one place using those. Still not that many (about 20 in current `main`), so someone interested could go through all of them.
